WeScan/Extensions/UIImage+Utils.swift

```diff
@@ -35,4 +40,50 @@ extension UIImage {
         return UIImage(cgImage: croppedImage)
     }
     
+    /// Scales the image to the specified size in the RGB color space.
+    ///
+    /// - Parameters:
+    ///   - scaleFactor: Factor by which the image should be scaled.
+    /// - Returns: The scaled image.
+    func scaledImage(scaleFactor: CGFloat) -> UIImage? {
+        guard let cgImage = self.cgImage else { return nil }
+        
+        let customColorSpace = CGColorSpaceCreateDeviceRGB()
+        
+        let width = CGFloat(cgImage.width) * scaleFactor
+        let height = CGFloat(cgImage.height) * scaleFactor
+        let bitsPerComponent = cgImage.bitsPerComponent
+        let bytesPerRow = cgImage.bytesPerRow
+        let bitmapInfo = cgImage.bitmapInfo.rawValue
+        
+        guard let context = CGContext(data: nil, width: Int(width), height: Int(height), bitsPerComponent: bitsPerComponent, bytesPerRow: bytesPerRow, space: customColorSpace, bitmapInfo: bitmapInfo) else { return nil }
```

Comment:
Why not use `UIGraphicsBeginImageContext`?

Reply:
From what I understand, `UIGraphicsBeginImageContext` is just a wrapper over this - and should have a negligible difference in performance, whether positive or negative. However, this allows us to select the colorSpace and if, for example, we had to change the scale or something in the future it gives us more control.